File: packages/charity-frontend/app/core/model/payment-info.ts
import {TransactionProvider} from "@/app/core/model/donation";

export enum PaymentOpenType {
    MODAL = 'MODAL',
    HREF = 'HREF',

}

interface PaymentInfo {
    url: string;
    openType: PaymentOpenType;
    provider: TransactionProvider;
    confirmMessage?: string;
    meta?: any;
}

export default PaymentInfo;

File: packages/charity/src/main/java/vn/edu/funix/charity/features/payment/entity/PaymentInfo.java
package vn.edu.funix.charity.features.payment.entity;

import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import lombok.AllArgsConstructor;
import lombok.Getter;
import vn.edu.funix.charity.entity.enumerate.TransactionProvider;

import java.util.Map;

@Getter
@AllArgsConstructor
public class PaymentInfo {
    private String url;

    @Enumerated(EnumType.STRING)
    private PaymentInfoOpenType openType;

    @Enumerated(EnumType.STRING)
    private TransactionProvider provider;

    private String confirmMessage;

    Map<String, Object> meta;
    public enum PaymentInfoOpenType {
        MODAL, HREF,
    }
}
